Feature/Level-Menu

timjacksonm opened the pull request:
Level Menu.
Includes button navigation. Back button & Home button. Might not need both - just unsure when and how this component will be accessed.
User can select a difficulty and the container will display levels 1-10 with anchor links to go to that page.
Still needs localstorage setup to display correct badge data. Currently have Badge1, Badge2, Badge3 included as default for first 3 levels as a template to showcase looks. Let me know what you think.

timjacksonm committed:
Added blue team dino svgs to legend.

timjacksonm committed:
created level 1 object data, added a new dino type with value of three for each team, adjusted legend to receive level data and display appropriate dinos and their values.

timjacksonm committed:
level updates
-Moved level data object into its own file.
-Added levelData state which determines the level that should be played based on url.
-Adjusted redDinos/blueDinos variables to include levelData. dinoType is the svg image, qty for quantity of that dinoType to put on the field. Removed styling and made a top level variable placeRandom
-Adjusted the jsx to map the dinos to field using levelData.

timjacksonm committed:
refactored the level data based on team feedback.

timjacksonm committed:
updated level data.
-created first 10 levels for beginner difficulty. first 3 levels the winning side is pre determined. After lvl 3 it is random and the legend board is always different from the previous level.

timjacksonm committed:
made more random again

timjacksonm committed:
adjusted naming of props passed to legend.

timjacksonm committed:
level-menu features
-added difficulty setting carousel for beginner, average, hard settings.
-ceated cards for the levels.
-added back button & home button.

timjacksonm committed:
moved level data into its own file.
-Level.js now handles rendering the cards for the level data.
-sets correct badge based on data the component is using.
-removed duplicates and adding a map function.

timjacksonm committed:
fixes
-removed unnessary parentheses on return
-capitalized I in incomplete.

timjacksonm committed:
removed height styles on some elements.

timjacksonm committed:
fixes
-adjusted styling on diffuculty selector. Now it doesn't bounce around due to the character length of a word. added fixe width to center text.
-added default data for beginner, average, hard difficutly. LevelContainer re-renders correct data based on center Text.
-added href attributes to buttonNavigation. Will update once we have pages figured out.

timjacksonm committed:
destructured set to [dinoId, points] to better explain what set is

timjacksonm committed:
changed teamId to legend.

timjacksonm committed:
updated to be 1 intead of :1. also updated index.js for levels to reflect changes

timjacksonm committed:
changed levels[levelId] to levels[difficulty] and removed previous commit changes.

timjacksonm committed:
refactored dino arrays. installed shuffle-array

timjacksonm committed:
changed require to import statement for shuffle.

timjacksonm committed:
changed getRange function to use reduce method.

timjacksonm committed:
changed 5-point-blue.svg to a different dino svg file. Renamed all svgs to correspond with their id in dinos. Re-wrote imports.

timjacksonm committed:
folder structure
-removed staticLevel folder and added data too src/data.
-moved levels, legends to their own js files inside of src/data.
-renamed index.js to dinos.js and left the dinos variable in the file.
-adjusted imports on level index.js and legend index.js

timjacksonm committed:
updated router for difficulty/stage. enabled prettier with double quote jsx. single quote everything else.

timjacksonm committed:
data updates
-added legend boards that include 4 dinos and also 3 dinos & operand. Have it as subtraction for now.
-added dino11-dino15 for dead dinos to be used as operands.
-created 10 levels for average and hard diffuclty. hard lvl 10 is kind of crazy. It is possible to have a negative value I think so probably won't work. We can change this as we get the rest of the game developed.

timjacksonm committed:
data more random

timjacksonm committed:
Changes
-Changed component / folder name from LevelMenu too LevelSelect.
-Lowercase the difficulty url links. Hard to hard for exmaple.
-Removed : in Link.

timjacksonm committed:
Changed all badge icons to incomplete default for now.

timjacksonm committed:
updated route path from /levelselect too /levels.

timjacksonm committed:
removed # symbol as it isn't needed.

timjacksonm committed:
merged upstream with main.

timjacksonm committed:
deleted dino svgs since they didn't match upstream.

timjacksonm committed:
merged main into branch. removed styled component Link and am now using Link from react-router-dom. Updated link to={path} to not include # symbol.

timjacksonm committed:
Link & styling Updates.
-Created AppLink styled component for the back and home buttons.
-Changed Link in levels.js to CardLink. The reason why I dont have this the same as AppLink is because the link needs to overlap the entire card container and be z-index 1 so that you can click anywhere on the card and proceed to the link.
-styles for both new components.

rankoliang commented on src/components/LevelSelect/Levels.js:
```diff
@@ -0,0 +1,83 @@
+  });
+  const [levels, setLevels] = useState(Object.entries(data[props.selection]));
+  const renderStatus = (status) => {
+    if (status === 'Incomplete') {
```
Better to use a switch/case statement here.

rankoliang commented on src/components/LevelSelect/Levels.js:
```diff
@@ -0,0 +1,83 @@
+    if (status === 'Incomplete') {
+      return <Incomplete />;
+    }
+    if (status === 'Badge1') {
```
Could be slightly more descriptive here, like `BlueComplete`, `RedComplete`, `Complete`

rankoliang reviewed (approved):
I have some minor things that you can change, but it looks good to me! Great work, I really like it. I think we could make it a modal later and have it pop up on the home screen, but it's not critical.

timjacksonm committed:
Merge remote-tracking branch 'upstream/main'

timjacksonm committed:
Merge branch 'main' into feature/Level-Menu

timjacksonm committed:
code updates
-Updated if statement to be switch statement.
-Changed badge component names to be more descriptive.
-Changed StyledLevel to BackgroundStyle as this page isnt a level.
-Updated svg file names to match new component names.

timjacksonm committed:
Updated svg file names to match new component names.

timjacksonm committed:
disabled the back button for now. As the level menu is only available from the Home screen. If we add it to the result page we can add a back button option to get back to the result screen.

timjacksonm commented:
Did a few more updates. Ready to be merged with new approval.

rankoliang reviewed (approved):
Looks good to me! 🚀
